Content.Server/Singularity/EntitySystems/ContainmentFieldGeneratorSystem.cs

```diff
@@ -169,7 +170,8 @@ private void RemoveConnections(Entity<ContainmentFieldGeneratorComponent> genera
         ChangeOnLightVisualizer(generator);
         ChangeFieldVisualizer(generator);
         _adminLogger.Add(LogType.FieldGeneration, LogImpact.Medium, $"{ToPrettyString(uid)} lost field connections"); // Ideally LogImpact would depend on if there is a singulo nearby
-        _popupSystem.PopupEntity(Loc.GetString("comp-containment-disconnected"), uid, PopupType.LargeCaution);
+        if(WasConnected)
+          _popupSystem.PopupEntity(Loc.GetString("comp-containment-disconnected"), uid, PopupType.LargeCaution);
```

Comment:
just move `component.IsConnected = false` to after this line instead of adding the extra variable

Reply:
Thank you original gangster. 

I was mistaken and believed that IsConnected was set to false within the loop, but that obviously is not the case.